bedrock/careers/views.py

```diff
@@ -62,10 +64,17 @@ class BenefitsView(L10nTemplateView):
 
 
 class PositionListView(LangFilesMixin, RequireSafeMixin, ListView):
-    queryset = Position.objects.exclude(job_locations="Remote")
     template_name = "careers/listings.html"
     context_object_name = "positions"
 
+    def get_queryset(self):
+        _key = "careers_position_listing_qs"
+        qs = cache.get(_key)
+        if qs is None:
+            qs = Position.objects.exclude(job_locations="Remote")
+            cache.set(_key, qs, settings.CACHE_TIME_SHORT)
+        return qs
```

Comment:
Should we cache the queryset or cache the list of objects?

Reply:
Ideally yeah, we'd cache the objects, not the qs, but it's a bit more fiddly than I expected.

Because these are CBVs, cacheing the queryset in `get_queryset` is easier to do (and to discover) than cacheing the `Position` objects themselves, because there's no clean way to slot in cacheing when the view [sets the `object_list` attribute](https://github.com/django/django/blob/857b1048d53ebf5fc5581c110e85c212b81ca83a/django/views/generic/list.py#L158) that I can see  

Later, even though we don't use pagination in these pages, the list view runs the data through [`paginate_queryset()`](https://github.com/django/django/blob/857b1048d53ebf5fc5581c110e85c212b81ca83a/django/views/generic/list.py#L53) which currently (Django 4.2.x) takes a `queryset` arg but treats it as a general iterable -- all of which means we _could_ make `get_queryset` return a list of objects (from the cache), not a `QuerySet`, but it kind of muddies things.

What do you think?

Reply:
Thanks for sharing that. It's a bit more complex than first glance. I'm actually a bit surprised we can cache the queryset objects, which seems more complex. I suppose they get pickled? But this is fine with me, I was mostly curious about your thoughts on the consideration. Thanks!